packages/endpoint/package.json

```diff
@@ -25,23 +25,40 @@
       ]
     }
   },
+  "exports": {
+    ".": [
+      {
+        "node": {
+          "import": "./node.mjs",
+          "require": "./dist/index.cjs"
+        },
+        "default": "./lib/index.js"
+      },
+      "./lib/index.js"
+    ],
+    "./package.json": "./package.json"
+  },
+  "type": "module",
```

Comment:
i strongly discourage this one - type module achieves nothing except breaking a bunch of tools, and making `.js` mean ESM - which will mean your "default" entry point is ESM, not CJS, and thus can't be required.

All ESM should only be `.mjs`, and `.js` should continue to only mean CJS. Using `.cjs` is "fine" but it will break in unexpected ways in older nodes, and it's not really worth the hassle.
```suggestion
```

Reply:
that reminds me to update the engine entry

Reply:
That doesn't change that plenty of tools - that are not node - will still break with this. Using it adds nothing except a minor aesthetic change (arguably a bad one).

Reply:
What tools? The tools this library cares about in order are:

webpack, node, react native, rollup, parcel, snowpack.

This design is optimized for webpack. Though you may disagree with webpack's decisions, that is the primary target. Supporting even tools that aren't primary are also desirable so long as the cost is reasonable.

I'd love to know more about the other tools to evaluate their likelihood of interacting with this library, along with precisely what sorts of things must be considered.

Reply:
Without knowing specifics - the application to the general is this: .js has existed long before node exits, and still exists outside that realm. Thus, all modern node tools should be aware of type specifier, and thus supporting modern node (legacy node is explicitly marked as not supported) is handled by this specification. This enables maximum compatibility with all tools not related or aware of node - which again is the more primary target of this library.

Reply:
Many tools look for *.js files but not necessarily *.mjs - especially legacy ones.